RealmTasks Apple/RealmTasks iOS/CellPresenter.swift

```diff
@@ -32,35 +32,37 @@ class CellPresenter<Item: Object where Item: CellPresentable> {
     }
 
     func deleteItem(item: Item) {
-        viewController.uiWrite {
-            guard !(item as Object).invalidated, let index = items.indexOf(item) else {
+        try! items.realm?.write {
+            guard !(item as Object).isInvalidated, let index = items.index(of: item) else {
                 return
             }
 
             items.realm?.delete(item)
-            viewController.tableView.deleteRowsAtIndexPaths([NSIndexPath(forRow: index, inSection: 0)], withRowAnimation: .Left)
+            viewController.tableView.deleteRows(at: [IndexPath(row: index, section: 0)], with: .left)
+            viewController.didUpdateList(reload: true)
         }
     }
 
     func completeItem(item: Item) {
-        viewController.uiWrite {
-            guard !(item as Object).invalidated, let index = items.indexOf(item) else {
+        try! items.realm?.write {
+            guard !(item as Object).isInvalidated, let index = items.index(of: item) else {
                 return
             }
 
-            let sourceIndexPath = NSIndexPath(forRow: index, inSection: 0)
-            let destinationIndexPath: NSIndexPath
+            let sourceIndexPath = IndexPath(row: index, section: 0)
+            let destinationIndexPath: IndexPath
             if item.completed {
                 // move cell to bottom
-                destinationIndexPath = NSIndexPath(forRow: items.count - 1, inSection: 0)
+                destinationIndexPath = IndexPath(row: items.count - 1, section: 0)
             } else {
                 // move cell just above the first completed item
                 let completedCount = items.filter("completed = true").count
-                destinationIndexPath = NSIndexPath(forRow: items.count - completedCount - 1, inSection: 0)
+                destinationIndexPath = IndexPath(row: items.count - completedCount - 1, section: 0)
             }
 
             items.move(from: sourceIndexPath.row, to: destinationIndexPath.row)
-            viewController.tableView.moveRowAtIndexPath(sourceIndexPath, toIndexPath: destinationIndexPath)
+            viewController.tableView.moveRow(at: sourceIndexPath, to: destinationIndexPath)
+            viewController.didUpdateList(reload: false)
```

Comment:
This block of code is already inside a `uiWrite` block. Is it necessary to encapsulate it again?